python/valis/cache.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Filename: main.py
+# Project: app
+# Author: José Sánchez-Gallego
+# Created: Monday, 9th December 2024
+# License: BSD 3-clause "New" or "Revised" License
+# Copyright (c) 2020 José Sánchez-Gallego
+# Last Modified: Monday, 9th December 2024
+# Modified By: José Sánchez-Gallego
+
+from __future__ import annotations
+
+import hashlib
+import json
+import logging
+from contextlib import asynccontextmanager
+from functools import wraps
+from inspect import Parameter, isawaitable, iscoroutinefunction
+from typing import (
+    TYPE_CHECKING,
+    Awaitable,
+    Callable,
+    List,
+    Optional,
+    ParamSpec,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast
+)
+
+from fastapi.dependencies.utils import (
+    get_typed_return_annotation,
+    get_typed_signature
+)
+from fastapi_cache import Backend, FastAPICache
+from fastapi_cache.backends.inmemory import InMemoryBackend
+from fastapi_cache.backends.redis import RedisBackend
+from fastapi_cache.decorator import _augment_signature, _locate_param
+from redis.asyncio.client import Redis
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.status import HTTP_304_NOT_MODIFIED
+
+from valis.settings import settings
+
+
+if TYPE_CHECKING:
+    from typing import AsyncIterator
+
+    from fastapi import FastAPI
+    from fastapi_cache.coder import Coder
+    from fastapi_cache.types import KeyBuilder
+
+
+__all__ = ['valis_cache', 'lifespan', 'valis_cache_key_builder']
+
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+logger = logging.getLogger("uvicorn.error")
+
+CACHE_TTL: float = 15_552_000 # 6 months
```

Comment:
Can we make this an attribute in the settings?  That way we can potentially change the cache ttl during deployment, or control it via an environment variable.   Does this number really use a format with `_`?  That seems odd.

Reply:
Done. I removed the _. Numbers in Python can use _ as thousands separator (the compiler ignores the underscores) but it's probably confusing.